lib/tunestakeoutwrapper.rb

```diff
@@ -0,0 +1,53 @@
+# this is the one that is talking to Charles API
+require 'httparty'
+# module
+module TunesTakeout
+  BASE_URL = "https://tunes-takeout-api.herokuapp.com/"
+  attr_reader :search, :top_favorites_by, :final_results
+
+  def self.search(query, limit = 10, seed = query)
+    # /v1/suggestions/search?query=banana
+    # find a tune with the passed query
+    data = HTTParty.get(BASE_URL + "v1/suggestions/search?query=#{query}&limit=#{limit}&seed=#{query}").parsed_response
+  end
+
+  def self.top_favorites_by
+    # hard coded limit of 20, possible refractor would be to make this an argument.
+    data = HTTParty.get(BASE_URL + "v1/suggestions/top?limit=20").parsed_response
+  end
+
+  def self.final_results(suggestions_hash)
+    # sending an id and retrieving the pairings for search
+    @final_results = []
```

Comment:
Instance variables are for an instance of the class, and therefore used on instance methods. These are class methods, the ones starting in `self`. You can change all of these to local variables.